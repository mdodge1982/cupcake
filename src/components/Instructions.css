.small {
	font-size: .8em;
}
@media screen and ( max-height: 500px ){
	.small {
		display: none;
	}
}
