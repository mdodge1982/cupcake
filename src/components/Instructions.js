import React, {Component} from 'react';
import './Instructions.css';

class Instructions extends Component {
	render() {
		let className="Instructions";
		let hint = '';
		if(this.props.small){
			className += ' small';
			hint = (
				<strong>HINT:</strong>
			);
		}
		return (
			<p className={className}>
				{hint} Use the <strong>left</strong> and <strong>right</strong> arrow keys to move your pastry bag,
				and the <strong>spacebar</strong> (or click anywhere on the board) to shoot frosting.
			</p>
		);
	}
}

export default Instructions;
