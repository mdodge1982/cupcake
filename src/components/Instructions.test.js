import React from 'react';
import renderer from 'react-test-renderer';
import Instructions from './Instructions';


test('Text changes when rendered small', () => {
	const component = renderer.create(
		<Instructions />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.small = true;
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
