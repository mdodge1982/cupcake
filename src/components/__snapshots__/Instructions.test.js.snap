// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text changes when rendered small 1`] = `
<p
  className="Instructions"
>
  
   Use the 
  <strong>
    left
  </strong>
   and 
  <strong>
    right
  </strong>
   arrow keys to move your pastry bag, and the 
  <strong>
    spacebar
  </strong>
   (or click anywhere on the board) to shoot frosting.
</p>
`;

exports[`Text changes when rendered small 2`] = `
<p
  className="Instructions"
>
  
   Use the 
  <strong>
    left
  </strong>
   and 
  <strong>
    right
  </strong>
   arrow keys to move your pastry bag, and the 
  <strong>
    spacebar
  </strong>
   (or click anywhere on the board) to shoot frosting.
</p>
`;
